node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -391,6 +382,18 @@ case class DataMessageHandler(
               }
             }
 
+        getHeadersF.recoverWith {
```

Comment:
For any invalid header, `chainApi` recovers and sends re-queries for headers. This future is then chained to to `getHeadersF` which again asks for headers. Two similar header requests are sent, the first one is processed the second one is invalid, again causing 2 requests.

Reply:
A bug introduced by yours truely, sorry about that.

Reply:
Not really :sweat_smile:, this was present before your refactor too.